core/pta/bcm/bnxt.c

```diff
@@ -45,6 +48,58 @@ static void close_session(void *session_context __unused)
 	DMSG("close entry point for static ta \"%s\"", BNXT_TA_NAME);
 }
 
+static TEE_Result get_bnxt_status(uint32_t type, TEE_Param p[TEE_NUM_PARAMS])
+{
+	if (type != TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_OUTPUT,
+				    TEE_PARAM_TYPE_NONE,
+				    TEE_PARAM_TYPE_NONE,
+				    TEE_PARAM_TYPE_NONE))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	p[0].value.a = bnxt_health_status();
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result get_bnxt_handshake_status(uint32_t type,
+					    TEE_Param p[TEE_NUM_PARAMS])
+{
+	if (type != TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INOUT,
+				    TEE_PARAM_TYPE_NONE,
+				    TEE_PARAM_TYPE_NONE,
+				    TEE_PARAM_TYPE_NONE))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	p[0].value.a = bnxt_wait_handshake(p[0].value.a);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result copy_bnxt_crash_dump(uint32_t types,
+				       TEE_Param params[TEE_NUM_PARAMS])
+{
+	uint32_t *d = NULL;
+	TEE_Result res = TEE_SUCCESS;
+
+	if (types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INOUT,
+				     TEE_PARAM_TYPE_VALUE_INPUT,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE)) {
+		DMSG("bad parameters types: 0x%" PRIx32, types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	d = (uint32_t *)params[0].memref.buffer;
+
+	if (!d)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	bnxt_copy_crash_dump((uint8_t *)d, params[1].value.a,
+			     params[1].value.b);
```

Comment:
Wait! If I understand correctly `a` and `b` are an offset and a size, but nowhere you check that the memory is accessible to the caller.

Reply:
Yes, The checks are implemented inside `bnxt_copy_crash_dump` API. 
I need to take care of the return value.

Reply:
`d` is not checked in relation with offset and size.

Reply:
Added check for memref.size against requested size

Reply:
OK LGTM